tests/e2e/csi_cns_telemetry_vc_reboot.go

```diff
@@ -0,0 +1,516 @@
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Waiting for claim to be in bound phase")
+		pvc, err := fpv.WaitForPVClaimBoundPhase(client, []*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(pvc).NotTo(gomega.BeEmpty())
+		pv := getPvFromClaim(client, pvclaim.Namespace, pvclaim.Name)
+		volumeID := pv.Spec.CSI.VolumeHandle
+
+		if guestCluster {
+			// svcPVCName refers to PVC Name in the supervisor cluster
+			svcPVCName = volumeID
+			volumeID = getVolumeIDFromSupervisorCluster(svcPVCName)
+			gomega.Expect(volumeID).NotTo(gomega.BeEmpty())
+		}
+
+		ginkgo.By("Creating pod to attach PV to the node")
+		pod, err := createPod(client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim}, false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		var vmUUID string
+		var vmUUID2 string
+		var exists bool
+		nodeName := pod.Spec.NodeName
+		if supervisorCluster {
+			annotations := pod.Annotations
+			vmUUID, exists = annotations[vmUUIDLabel]
+			gomega.Expect(exists).To(gomega.BeTrue(), fmt.Sprintf("Pod doesn't have %s annotation", vmUUIDLabel))
+			_, err := e2eVSphere.getVMByUUID(ctx, vmUUID)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		if guestCluster {
+			vmUUID, err = getVMUUIDFromNodeName(nodeName)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		if vanillaCluster {
+			vmUUID = getNodeUUID(client, pod.Spec.NodeName)
+		}
+
+		ginkgo.By(fmt.Sprintf("Verify volume: %s is attached to the node: %s", volumeID, nodeName))
+		isDiskAttached, err := e2eVSphere.isVolumeAttachedToVM(client, volumeID, vmUUID)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(isDiskAttached).To(gomega.BeTrue(), "Volume is not attached to the node")
+
+		ginkgo.By("Rebooting VC")
+		vcAddress := e2eVSphere.Config.Global.VCenterHostname + ":" + sshdPort
+		err = invokeVCenterReboot(vcAddress)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Sleep for a short while for the VC to shutdown, before invoking PVC creation and cluster-distribution value
+		time.Sleep(pollTimeoutShort)
+
+		ginkgo.By("Set cluster-distribution value while the VC reboot in progress")
+		// Set Cluster-distribution while the VC reboot in progress
+		setClusterDistribution(ctx, client, vanillaClusterDistributionWithSpecialChar)
+
+		defer func() {
+			//Reset the cluster distribution value to default value "CSI-Vanilla"
+			setClusterDistribution(ctx, client, vanillaClusterDistribution)
+		}()
+
+		// decide which test setup is available to run
+		if vanillaCluster {
+			ginkgo.By("Creating another PVC for Vanilla setup")
+			storageclass2, pvclaim2, err = createPVCAndStorageClass(client, namespace, nil, nil, "", nil, "", false, "")
+		} else if supervisorCluster {
```

Comment:
Since this test is not running for SVC or GC. Please remove this block or Add a TODO saying this block is not getting used

Reply:
This comment applies for all the places where ever guest or supervisor cluster's code are present.